docs/source/generics.rst

```diff
@@ -223,20 +223,15 @@ example we use the same type variable in two generic functions:
    def last(seq: Sequence[T]) -> T:
        return seq[-1]
 
-Note that a variable cannot have a type variable in its type unless the type variable is bound in a containing generic class or function:
+A variable cannot have a type variable in its type unless the type
+variable is bound in a containing generic class or function:
```

Comment:
Hm, sadly I just realized this is not quite correct, since this is possible:
```py
T = TypeVar('T')
S = TypeVar('S')

def repeat(n: int) -> Callable[[Callable[[T], T]], Callable[[T], T]]:
    def outer(func: Callable[[S], S]) -> Callable[[S], S]:
        def inner(x: S) -> S:
            for i in range(n):
                x = func(x)
            return x
        return inner
    return outer

fff = repeat(3)
reveal_type(fff)  # 'def [T] (def (T`-1) -> T`-1) -> def (T`-1) -> T`-1'
```
But I don't want to include such an example here -- I just wish to find a way to write it that doesn't exclude it.

Also this much simpler example *should* be invalid but isn't:
```py
def foo() -> T:
    pass

a = foo()
```

So I'm not sure what to do. Maybe @ilevkivskyi has an idea?

Reply:
`def foo() -> T: pass`

seems to be an unidiomatic way of writing

`def foo() -> Any: pass`

Reply:
They don't mean the same thing in mypy. The former returns a type containing `<nothing>` (meaning it has no values).